src/schema/schema.ts

```diff
@@ -0,0 +1,41 @@
+import { boolean, mixed, number, object, ref, string } from 'yup';
+
+export const schema = object({
+  name: string()
+    .required('поле обязательно')
+    .test(
+      'Заглавная буква',
+      'введите имя с заглавной буквы',
+      (value) => !!value && value[0] === value[0].toUpperCase()

```

Comment:
This is not correct check for the first UpperCase letter. User can use numbers for name.
RegExp will be great here.